ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+    int parsedNumHeaders;
+};
+
+type RowValues record {
+    map<json>[] c;
+};
+
+# The client used by the generated persist clients to abstract and
+# execute API calls that are required to perform CRUD operations.
+public client class GoogleSheetsClient {
+
+    private final sheets:Client googleSheetClient;
+    private final http:Client httpClient;
+    private string spreadsheetId;
+    private int sheetId;
+    private string entityName;
+    private string tableName;
+    private string range;
+    private map<SheetFieldMetadata> fieldMetadata;
+    private map<string> dataTypes;
+    private string[] keyFields;
+    private function (string[]) returns stream<record {}, Error?>|Error query;
+    private function (anydata) returns record {}|InvalidKeyError queryOne;
+    private map<function (record {}, string[]) returns record {}[]|error> associationsMethods;
+
+    # Initializes the `GSheetClient`.
+    #
+    # + GSheetClient - The `sheets:Client`, which is used to execute google sheets operations
+    # + httpClient - The `http:Client`, which is used to execute http requests
+    # + metadata - Metadata of the entity
+    # + spreadsheetId - Id of the spreadsheet
+    # + sheetId - Id of the sheet
+    # + return - A `persist:Error` if the client creation fails
+    public function init(sheets:Client googleSheetClient, http:Client httpClient, SheetMetadata sheetMetadata, string spreadsheetId, int sheetId) returns Error? {
+        self.entityName = sheetMetadata.entityName;
+        self.spreadsheetId = spreadsheetId;
+        self.tableName = sheetMetadata.tableName;
+        self.fieldMetadata = sheetMetadata.fieldMetadata;
+        self.range = sheetMetadata.range;
+        self.httpClient = httpClient;
+        self.keyFields = sheetMetadata.keyFields;
+        self.googleSheetClient = googleSheetClient;
+        self.dataTypes = sheetMetadata.dataTypes;
+        self.query = sheetMetadata.query;
+        self.queryOne = sheetMetadata.queryOne;
+        self.associationsMethods = sheetMetadata.associationsMethods;
+        self.sheetId = sheetId;
+
+    }
+
+    # Performs an append operation to insert entity instances into a table.
+    #
+    # + insertRecords - The entity records to be inserted into the table
+    # + return - An `sql:ExecutionResult[]` containing the metadata of the query execution
+    # or a `persist:Error` if the operation fails
+    public isolated function runBatchInsertQuery(record {}[] insertRecords) returns Error? {
+        string[] fieldMetadataKeys = self.fieldMetadata.keys();
+        foreach record {} rowValues in insertRecords {
+            string metadataValue = self.generateMetadataValue(self.keyFields, rowValues);
+            sheets:DeveloperMetadataLookupFilter filter = {locationType: "ROW", metadataKey: self.tableName, metadataValue: metadataValue};
+            sheets:ValueRange[]|error output = self.googleSheetClient->getRowByDataFilter(self.spreadsheetId, self.sheetId, filter);
```

Comment:
What would happen here if `output` is an `error`? Seems like we don't handle that case.

Reply:
updated in [1873a0f](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/1873a0f9e51bc1a8608533d7c2719ca2f17f54c3)